website/project/utils.py

```diff
@@ -25,21 +25,16 @@ def serialize_node(*args, **kwargs):
 
 
 def recent_public_registrations(n=10):
+    from django.db.models import Q as DQ
     Registration = apps.get_model('osf.Registration')
-    registrations = Registration.find(
-        CONTENT_NODE_QUERY &
-        Q('is_public', 'eq', True)
-    ).get_roots().sort(
-        '-registered_date'
-    )
-    for reg in registrations:
-        if not n:
-            break
-        if reg.is_retracted or reg.is_pending_embargo:
-            # Filter based on calculated properties
-            continue
-        n -= 1
-        yield reg
+
+    return Registration.objects.filter(
+        is_public=True,
+        is_deleted=False,
+    ).filter(
+        DQ(DQ(embargo__isnull=True) | ~DQ(embargo__state='unapproved')) &
+        DQ(DQ(retraction__isnull=True) | ~DQ(retraction__state='approved'))
+    ).get_roots().order_by('-registered_date').limit(n)
```

Comment:
btw, don't let this be much of a blocker, its certainly an improvement and we'll see any nuance in newrelic.